src/worldstate/src/routines/Gate/PassGate.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Created By: Joel Ahn
+ * Created On: March 5th, 2018
+ * Description: World State FSM node. Checks to see if
+ *              robot still needs to pass gate.
+ */
+
+#include "routines/Gate/PassGate.h"
+
+void PassGate::setupNodeSubscriptions(ros::NodeHandle nh) {
+    std::string gateDetectTopic = "/gateDetect/output";
+    std::string imuDataTopic    = "/imu/is_calibrated";
+    gate_detect_listener = nh.subscribe(gateDetectTopic, 10, &PassGate::gateDetectCallBack, this);
+    //nh.subscribe("/imu/is_calibrated", 10, &PassGate::imuDataCallback, this);
+}
+
+void PassGate::sleep(){
+    gate_detect_listener.shutdown();
+}
+
+void PassGate::gateDetectCallBack(
+const gate_detect::gateDetectMsg::ConstPtr& msg) {
+    worldstate::StateMsg msg_to_publish;
+    msg_to_publish.state = worldstate::StateMsg::passingGate;
+
+    /*
+     * Assume for now if the gate disappears it has successfully
+     * passed through the gate
+     */
+    if (!msg->detectLeft && !msg->detectRight && msg->detectTop){
+        msg_to_publish.state = worldstate::StateMsg::locatingDie;
+    }
+
+    /*
+     * What are the failed states?
+     */
```

Comment:
I dunno what the failed states are, but we should probably get that figured out before merging this :stuck_out_tongue_winking_eye:

Reply:
Aside from the failed state where its veers off course, as measured by the IMU, or it doesn't meet the clearance height of the gate, I'm having trouble what it's look like in fish eye view.